sklearn/tests/test_naive_bayes.py

```diff
@@ -616,6 +659,80 @@ def test_cnb():
     assert_array_almost_equal(clf.feature_log_prob_, normed_weights)
 
 
+def test_catnb():
+    # Check the ability to predict the training set.
+    clf = CategoricalNB()
+    y_pred = clf.fit(X2, y2).predict(X2)
+    assert_array_equal(y_pred, y2)
+
+    X3 = np.array([[1, 4], [2, 5]])
+    y3 = np.array([1, 2])
+    clf = CategoricalNB(alpha=1, fit_prior=False)
+
+    # Check that unexpected parameters throw an error
+    clf.handle_unknown = 'asdf'
+    assert_raises(ValueError, clf.fit, X3, y3)
```

Comment:
Please always check the error/warning message.

Also, for test functions that you added, please use pytest's `with pytest.raises` and `with pytest.warns` decorators.

For pre-existing test functions that already use `assert_raises` or `assert_warns`, you can keep them as-is (but still check the messages please)